test/unit/mock/profiles/deprecation/resource_pack/libraries/deprecation_tester.rb

```diff
@@ -0,0 +1,62 @@
+
+require 'stringio'
+
+class DeprecationTester < Inspec.resource(1)
+  name :deprecation_tester
+
+  DEPRECATION_CFG = <<~EOC
+  {
+    "file_version": "1.0.0",
+    "unknown_group_action": "warn",
+    "groups": {
+      "a_group_that_will_warn" : { "action": "warn" },
+      "a_group_that_will_exit" : { "action": "exit"  },
+      "a_group_that_will_exit_with_a_code" : { "action": "exit", "exit_status": 8 },
+      "an_ignored_group" : { "action": "ignore" },
+      "a_group_that_will_fail" : { "action": "fail_control" }
+    }
+  }
+  EOC
+
+  def fail_me
+    Inspec::Deprecation::Deprecator.class_test_cfg_io(StringIO.new(DEPRECATION_CFG))
+
+    #deprecate(:a_group_that_will_fail, 'This should fail')
```

Comment:
Are these comments needed? I removed them https://github.com/inspec/inspec/pull/3725 but...want to make sure that was the right action.

Reply:
The background here is that the original issue asked for a new resource DSL method, `deprecate`; and did not extend to making the facility available throughout the codebase.

Once `Inspec.deprecate` was added, having bare `deprecate` seemed a bit magical, and was a distinction without a difference, so I didn't go that route.  The commented out lines were cruft, and should indeed be removed.